### ISSUE TITLE
Question about the html conversion used in fonduer

### ISSUE THREAD

**Issue by dolphin-zs** [HazyResearch/snorkel#809]
Dear all,

Fonduer (snorkel) is a dramatically easy framework for extracting document-level relations.

We have applied it into our project and find that Fonduer relies on two kinds of input: the original pdf file and the converted html file. As stated in [the manuscript](https://arxiv.org/abs/1703.05028), we need to employ tools, such as poppler, to convert .pdf file into .html file.

However, when we apply the same tool (poppler), we can not get the well-converted html file as the ones provided in Fonduer tutorials. Our converted html files often lost table information and it seems poppler just generates the html line by line following the layout of pdf. We also find that some proprietary tools can get better results, but they are often rechargeable.

So, can you please give us some guidances about the converter tools?

Thanks!

**Comment by bhancock8** [comment]
Hi @dolphin-zs,

Glad to hear you're making good use of Fonduer!

Your results are consistent with what we've seen as well. To do the full pdf to html conversion, we do use proprietary tools (we tried both ABBYY FineReader and Adobe Acrobat; the docs in the tutorial were produced by Acrobat). The poppler tools were used primarily to acquire bounding box information at the world level so that we could add a visual layer of representation to the words in the document.  Unfortunately, even the best proprietary pdf-to-html converters still make many mistakes, and we couldn't find any converter that did a satisfactory conversion job _and_ gave us bounding box information, so that's why we had to use poppler as well and then merge the results (i.e., from Adobe we got the textual, structural (html), and tabular (from the html) attributes, and from poppler we got the visual attributes).
